elements/geosearch/main.js

```diff
@@ -0,0 +1,375 @@
+  }
+
+  async onInput(e) {
+    this._query = e.target.value;
+
+    // Ignore requests with less than 2 characters since the API might respond with a 400 to them.
+    if (this._query.length <= 1) {
+      this._isListVisible = false;
+      return;
+    } else {
+      this._isListVisible = true;
+    }
+
+    let bounce = _debounce(async () => {
+      // Switch from mock data to API if `endpoint` and `key` are set and not empty.
+      if (
+        this.endpoint &&
+        this.key &&
+        this.endpoint.length > 0 &&
+        this.key.length > 0
+      ) {
+        var template = Handlebars.compile(this.endpoint);
+
+        // Execute the compiled template and store the resulting URL.
+        let url = template({
+          query: this._query,
+          key: this.key,
+          limit: this.limit,
+        });
+
+        await this.fetchRemoteData(url);
+      } else {
+        this.useMockData();
+        console.info("Using mock data for EOxGeoSearch");
+      }
+    }, this.interval);
+
+    bounce();
+  }
+
+  onInputBlur() {
+    this._isInputVisible = false;
+    this._isListVisible = false;
+    this._query = "";
+  }
+
+  onButtonClick() {
+    this._isInputVisible = !this._isInputVisible;
+
+    // Auto-focus the input field when it becomes visible
+    if (this._isInputVisible) {
+      this.renderRoot.querySelector("#gazetteer").focus();
+    }
+  }
+
+  handleSelect(item) {
+    this._isListVisible = false;
+
+    this.onSelect(item);
+  }
+
+  getFlexDirection() {
+    return this.direction === "up"
+      ? "column-reverse"
+      : this.direction === "left"
+      ? "row-reverse"
+      : this.direction === "down"
+      ? "column"
+      : this.direction === "right"
+      ? "row"
+      : "row";
+  }
+
+  getResultsDirection() {
+    return this.resultsDirection === "up"
+      ? "column-reverse"
+      : this.resultsDirection === "left"
+      ? "row-reverse"
+      : this.resultsDirection === "down"
+      ? "column"
+      : this.resultsDirection === "right"
+      ? "row"
+      : "row";
+  }
+
+  getVerticalAlign() {
+    return this.resultsDirection === "up" ? "end" : "start";
+  }
+
+  getMarginDirection(direction) {
+    return direction === "up"
+      ? "top"
+      : direction === "left"
+      ? "left"
+      : direction === "down"
+      ? "bottom"
+      : direction === "right"
+      ? "right"
+      : "row";
+  }
+
+  render() {
+    return html`
+      <style>
+        ${button}
+      </style>
+      <div
+        class="geosearch ${this.small ? "small" : ""}"
+        style="
+            flex-direction: ${this.getFlexDirection()};
+            align-items: ${this.getVerticalAlign()};
+        "
+      >
+        <button
+          class="${this.button ? "" : "hidden"}"
+          style="
+                        margin-${this.getMarginDirection(this.direction)}: 12px;
+                        background: ${this._isInputVisible
+            ? "#0078CE"
+            : "#004170"};
```

Comment:
It would be necessary to allow the parent application to override the styling via CSS variables, here the color seems to be pretty hardcoded.